### PR TITLE
Drop ansible-runner repository management

### PR DESCRIPTION
**Pull request by ehelms**
The ansible-runner package is now delivered via the Foreman plugins repository.

Requires that ansible-runner be in the plugins repository (https://github.com/theforeman/foreman-packaging/pull/8434)

### CODE REVIEW

**Comment by evgeni** [comment]
Should we clean up the old repo definition?

**Comment by evgeni** [comment]
And speaking if cleanup, I think f-maintain also has some (wrong) special handling that should be removed.

**Comment by evgeni** [comment]
> > And speaking if cleanup, I think f-maintain also has some (wrong) special handling that should be removed.
> 
> I grepped around and couldn't find any. Anything in particular you are thinking it did?

https://github.com/theforeman/foreman_maintain/blob/5f9bf72d5d09f6277d5ee581553f9a7273d8223f/definitions/checks/non_rh_packages.rb#L31

**Comment by ehelms** [comment]
> https://github.com/theforeman/foreman_maintain/blob/5f9bf72d5d09f6277d5ee581553f9a7273d8223f/definitions/checks/non_rh_packages.rb#L31

I think we still want that? As we deliver ansible-runner in upstream and downstream does as well?

**Comment by evgeni** [comment]
No, that check is to allow upstream runner packages in downstream, which is wrong.

**Comment by ehelms** [comment]
> No, that check is to allow upstream runner packages in downstream, which is wrong.

I still don't think I understand. The only thing we are changing is the version of ansible-runner being packaged up. Why would that affect this logic?

**Review by ekohl — COMMENTED** [review]
The code reads well, but there is the part of compatibility. This is breaking, unless we backport the packaging changes to 3.4 (which I'd actually like to see). See https://github.com/theforeman/puppet-foreman_proxy#compatibility as well.

Looks like for now we don't have a breaking change, though https://github.com/theforeman/puppet-foreman_proxy/pull/630 will be. If we do go for the breaking change route, I'd like to get a release out with the current changes first.

File: manifests/plugin/ansible/runner.pp [hunk 1]
```diff
@@ -1,28 +1,15 @@
...
-          gpgcheck => true,
-          gpgkey   => 'https://releases.ansible.com/keys/RPM-GPG-KEY-ansible-release.pub',
-          enabled  => '1',
-          before   => Package[$package_name],
-        }
-      }
-      default: {
-        fail("Repository containing 'ansible-runner' not known for '${facts['os']['family']}'")
-      }
+
```
Comment (ekohl) [inline manifests/plugin/ansible/runner.pp:8]: I think this breaks lint
```suggestion
```

**Comment by ekohl** [comment]
> I still don't think I understand. The only thing we are changing is the version of ansible-runner being packaged up. Why would that affect this logic?

In foreman-maintain it verifies the vendor on packages, but allows an exception for ansible-runner. That exception becomes redundant once it's packaged by RH and can be dropped.

**Comment by ehelms** [comment]
> In foreman-maintain it verifies the vendor on packages, but allows an exception for ansible-runner. That exception becomes redundant once it's packaged by RH and can be dropped.

This is what confuses me - it's _already_ packaged by RH.

**Comment by ekohl** [comment]
The exclusion doesn't break anything, it just makes the check weaker than it needs to. So even without this, it can be dropped.

**Comment by evgeni** [comment]
> > In foreman-maintain it verifies the vendor on packages, but allows an exception for ansible-runner. That exception becomes redundant once it's packaged by RH and can be dropped.
> 
> This is what confuses me - it's _already_ packaged by RH.

And that's the wrong part. It allows the non RH (= upstream) package on an RH installs (because someone sometimes forgot to set manage_repo = false)

**Comment by ehelms** [comment]
> > > In foreman-maintain it verifies the vendor on packages, but allows an exception for ansible-runner. That exception becomes redundant once it's packaged by RH and can be dropped.
> > 
> > 
> > This is what confuses me - it's _already_ packaged by RH.
> 
> And that's the wrong part. It allows the non RH (= upstream) package on an RH installs (because someone sometimes forgot to set manage_repo = false)

Done over here -- https://github.com/theforeman/foreman_maintain/pull/644

**Comment by ehelms** [comment]
> Looks like for now we don't have a breaking change, though #630 will be. If we do go for the breaking change route, I'd like to get a release out with the current changes first.

We already have a merged breaking change (https://github.com/theforeman/puppet-foreman_proxy/pull/776). There are two non-breaking commits that if we wanted we'd have to do a branch and release with those:

 * https://github.com/theforeman/puppet-foreman_proxy/commit/41bbfbaf14d0ce0251e6cae3afebc1ab70b1f929
 * https://github.com/theforeman/puppet-foreman_proxy/commit/7ea5ed00413c23d7f3b1e236903efc9a2cf64d21

Given that would require a branch, I think a release is a separate question from whether this can go in the code base now. 

I have included an update to the README.

File: README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Part of the Foreman installer: <https://github.com/theforeman/foreman-installer>
 
 | Module version | Proxy versions | Notes                                               |
 |----------------|----------------|-----------------------------------------------------|
+| 24.x           | 3.5 and newer  |                                                     |
```
Comment (ekohl) [inline README.md:15]:
```suggestion
| 24.x           | 3.5 and newer  | See compatibility notes in its README for 3.1-3.4 |
```

File: README.md [hunk 3]
```diff
@@ -27,6 +28,7 @@ Part of the Foreman installer: <https://github.com/theforeman/foreman-installer>
 | 2.x            | 1.5 - 1.10     |                                                     |
 | 1.x            | 1.4 and older  |                                                     |
 
+ * 24.x dropped management of ansible-runner repository, ansible-runner is now in the Foreman plugin repository
```
Comment (ekohl) [inline README.md:31]:
```suggestion
 * 24.x dropped management of ansible-runner repository, ansible-runner is now in the Foreman plugin repository. This requires Foreman 3.5.
```

**Review by ekohl — COMMENTED** [review]
Small notes on the readme bit, otherwise :+1:

**Review by ekohl — APPROVED** [review]

**Pull request merged**